panic: runtime error: hash of unhashable type object.BigArray

repro:
```
keys(info.all_ids[0]) 
```

```
panic: runtime error: hash of unhashable type object.BigArray

goroutine 1 [running]:
grol.io/grol/eval.Cache.Get(0x14000110de0, {0x14000489f00?, 0x0?}, {0x140001057d0?, 0x1, 0x10474e5a8?})
	grol.io/grol/eval/memo.go:37 +0x120
grol.io/grol/eval.(*State).applyFunction(0x1400012e2a0, {0x14000113b58, 0x4}, {0x1047e2c60?, 0x140001913b0?}, {0x140001057d0, 0x1, 0x1})
	grol.io/grol/eval/eval.go:405 +0xe4
grol.io/grol/eval.(*State).evalInternal(0x1400012e2a0, {0x1047ce700?, 0x140003d6ed0})
	grol.io/grol/eval/eval.go:196 +0xb44
grol.io/grol/eval.(*State).Eval(0x1400012e2a0, {0x1047ce700?, 0x140003d6ed0?})
	grol.io/grol/eval/eval_api.go:99 +0x40
[...]
```

--- a/eval/eval_test.go
+++ b/eval/eval_test.go
@@ -948,3 +948,12 @@ func TestSmallMapSorting(t *testing.T) {
 		t.Errorf("wrong result, got %q expected %q", resStr, expected)
 	}
 }
+
+func TestCrashKeys(t *testing.T) {
+	inp := `keys(info.all_ids[0])`
+	s := eval.NewState()
+	_, err := eval.EvalString(s, inp, false)
+	if err != nil {
+		t.Errorf("should not have errored: %v", err)
+	}
+}

--- a/main_test.txtar
+++ b/main_test.txtar
@@ -192,6 +192,11 @@ stderr 'parser error'
 stderr 'parser error'
 stderr '@'
 
+# panic on keys() of large array
+grol -quiet -panic -c 'keys(info.all_ids[0])'
+!stderr panic
+
+
 -- sample_test.gr --
 // Sample file that our gorepl can interpret
 // <--- comments

--- a/object/object.go
+++ b/object/object.go
@@ -54,17 +54,26 @@ type Number interface {
 */
 
 // Hashable in tem of Go map for cache key.
-// TODO: small arrays/maps should be cacheable by putting them in a fixed array.
 func Hashable(o Object) bool {
 	switch o.Type() { //nolint:exhaustive // We have all the types that are hashable + default for the others.
 	case INTEGER, FLOAT, BOOLEAN, NIL, ERROR, STRING:
 		return true
 	case ARRAY:
-		if _, ok := o.(SmallArray); ok {
+		if sa, ok := o.(SmallArray); ok {
+			for _, el := range sa.smallArr[:sa.len] {
+				if !Hashable(el) {
+					return false
+				}
+			}
 			return true
 		}
 	case MAP:
-		if _, ok := o.(SmallMap); ok {
+		if sm, ok := o.(SmallMap); ok {
+			for _, kv := range sm.smallKV[:sm.len] {
+				if !Hashable(kv.Key) || !Hashable(kv.Value) {
+					return false
+				}
+			}
 			return true
 		}
 	}

